### DISCUSSION TITLE
Does Document Session handle disposing itself

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [ravendb/ravendb#16778] [Q&A] [answered]
We had code for a web application that would dispose of the document session after a request has finished. We recently upgraded our Raven Client API from 3.x to 5.x and noticed after startup and initializing a document session for a controller we got an error when we went to query using the document session that the session had been disposed. Do we need to have this functionality in the new version of Raven? Do we need to manually dispose of a document session?

**Comment by SbiCA** [comment]
@mfonnesbeck yes that's an application concern, I usually end up with the following line 
``` c#
 using var session = _documentStore.OpenAsyncSession();
```

Also I'm not aware of any feature of Raven Client doing auto dispose, of course eventually if GC kicks in but that's just the same as with any other `IDisposable`

**Comment by georgiosd** [comment, marked as answer]
Usually this ties into your dependency injection container, if the environment you're using has one.

If you're using ASP.NET Core for example, if you register a factory for the session as transient, it will handle disposing it at the end of the scope (request).
